lib/effects/move_effect.dart

```diff
@@ -39,34 +37,25 @@ class MoveEffect extends PositionComponentEffect {
     _xOriginal = component.x;
     _yOriginal = component.y;
 
-    _xDistance = _size(destination.x, component.x);
-    _yDistance = _size(destination.y, component.y);
+    _xDistance = _distance(destination.x, component.x);
+    _yDistance = _distance(destination.y, component.y);
 
     _xDirection = _direction(destination.x, component.x);
     _yDirection = _direction(destination.y, component.y);
 
-    _travelTime = max(
-      _xDistance / speed,
-      _yDistance / speed,
-    );
+    final totalDistance = sqrt(pow(_xDistance, 2) + pow(_yDistance, 2));
+    travelTime = totalDistance / speed;
   }
 
   @override
   void update(double dt) {
+    super.update(dt);
+
     if (!hasFinished()) {
-      final double percent = min(1.0, _elapsedTime / _travelTime);
-      final double c = curve?.transform(percent) ?? 1.0;
+      final double c = curve?.transform(percentage) ?? 1.0;
 
       component.x = _xOriginal + _xDistance * c * _xDirection;
       component.y = _yOriginal + _yDistance * c * _yDirection;
-    } else {
-      component.x = destination.x;
```

Comment:
This is necessary to make sure the position is correct when the effect has finished

Reply:
I think we can make sure that it ends up in the correct in the if clause, feels weird to set it after it should be finished?

Reply:
I don't think it is weird, but I also don't mind making it in the if clause, I just find it important to make sure, as when I first made the effects, in my tests, it was common to end up on the wrong position/size.

Reply:
This is fixed now.

Reply:
I can’t spot where we ensure that the final position is ensured, anything I am missing.

Reply:
Since hasFinished is based on the percentage done now `c` will always be 1.0 in here at some point and when that happens it will be at its destination. But good that you commented on this, realized `super.update(dt);` has to be made after the if block.